File: SIS104Menu/app/src/main/java/com/example/sis104menu/Ejercicios/Ecuacion.java
package com.example.sis104menu.Ejercicios;

public class Ecuacion {
    private  Double a;
    private  Double b;
    private  Double c;

    public Ecuacion(Double a, Double b, Double c) {
        this.a = a;
        this.b = b;
        this.c = c;
    }

    public Double getA() {
        return a;
    }

    public void setA(Double a) {
        this.a = a;
    }

    public Double getB() {
        return b;
    }

    public void setB(Double b) {
        this.b = b;
    }

    public Double getC() {
        return c;
    }

    public void setC(Double c) {
        this.c = c;
    }

    private Double D(){
        return  (this.b * this.b) - (4 * this.a * this.c);
    }

    public String Raices(){
        String solucion;
        if(D() == 0){
            Double x = - this.b /(2*this.a);
            solucion = "X1 = X2 = " + x;
        }else {
            if(D() > 0){
                Double x1 = (-this.b + Math.sqrt(D())/(2 * this.a));
                Double x2 = (-this.b - Math.sqrt(D())/(2 * this.a));
                solucion = "X1 = "+x1+"\n x2 = "+x2;
            }else {
                Double parteReal = -this.b/(2*this.a);
                Double parteImag = Math.sqrt(Math.abs(D()))/(2*this.a);
                solucion = "x1=" + parteReal+ "-"+parteImag + "\n x2 = "+parteReal+ "+" + parteImag + "i";
            }
        }

        return solucion;
    }
}

File: SIS104_Kotlin/app/src/main/java/com/example/sis104_kotlin/ejercicios/Ecuacion.kt
package com.example.sis104_kotlin.ejercicios

import kotlin.math.sqrt

class Ecuacion(a: Double, b: Double, c: Double) {

    private var _a: Double = 0.0
    private var _b: Double = 0.0
    private var _c: Double = 0.0

    init {
        _a = a
        _b = b
        _c = c
    }

    private fun d(): Double{
        return _b * _b - 4 * _a * _c
    }


    public  fun Raices(): String{
        var s: String = ""
        var x1: Double = 0.0
        var x2: Double = 0.0

        if(d() == 0.0){
            x1 = -_b /(2*_a)
            x2 = x1
            s = "Raices x1 = x2 = $x1"
        }else if(d() > 0.0){
            x1 = (-_b + sqrt(d()))/(2*_a)
            x2 = (-_b - sqrt(d()))/(2*_a)
            s = "Raices: x1 = $x1, x2=$x2"
        }else{
            var real = -_b/(2*_a)
            var img = sqrt(-d())/(2*_a)
            s = "raices: x1= $real + $img i, x2 = $real - $img"
        }

        return s
    }
}